File: node/director/battery-check/handler.test.js
'use strict'

const assert = require('node:assert/strict')
const { test } = require('node:test')

const handler = require('./handler')

test('raises an alert below the battery threshold', async () => {
  const context = new TestContext()
  await handler({ body: { battery_percent: 12 } }, context)

  assert.equal(context.statusCode, 200)
  assert.deepEqual(context.body, {
    value_percent: 12,
    threshold_percent: 20,
    alert: true
  })
})

class TestContext {
  status (value) {
    this.statusCode = value
    return this
  }

  headers (value) {
    this.headerValues = value
    return this
  }

  succeed (value) {
    this.body = value
    return value
  }
}

File: python/director/battery-check/handler_test.py
import json
from types import SimpleNamespace

try:
    from . import handler as h
except ImportError:
    import handler as h


def test_sets_alert_below_threshold():
    event = SimpleNamespace(body=json.dumps({"battery_percent": 12}).encode())
    response = h.handle(event, {})

    assert response["statusCode"] == 200
    assert response["body"]["alert"] is True
    assert response["body"]["threshold_percent"] == 20
